### PR TITLE
Upgrade to Babel 6

### PR DESCRIPTION
**Pull request by doug-wade**
This allows us to build lazyload with babel 6

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -260,5 +260,3 @@ LazyLoad.defaultProps = {
 };
 
 export default LazyLoad;
-
-export lazyload from './decorator';
```
Comment (doug-wade) [inline src/index.js]: As far as I can tell, the default key from this file (`export default LazyLoad`) is being overwritten by the default key being exported from `'./decorator'`.  By importing from `react-lazyload/decorator` directly, we don't have contention for the key.

File: package.json [hunk 2]
```diff
@@ -26,10 +26,15 @@
...
   "devDependencies": {
-    "babel": "~5.8.21",
-    "babel-core": "~5.8.21",
+    "babel": "~6.5.2",
+    "babel-cli": "^6.7.7",
+    "babel-core": "~6.7.7",
     "babel-eslint": "^6.0.2",
-    "babel-loader": "~5.3.2",
+    "babel-loader": "~6.2.4",
+    "babel-plugin-transform-decorators-legacy": "^1.3.4",
```
Comment (doug-wade) [inline package.json]: This is what is [recommended by Babel](http://babeljs.io/docs/plugins/transform-decorators/)

File: package.json [hunk 3]
```diff
@@ -4,11 +4,11 @@
   "description": "Lazyload your Component, Image or anything matters the performance.",
   "main": "lib/index.js",
   "scripts": {
-    "test": "./node_modules/karma/bin/karma start test/karma.conf.js",
-    "demo:watch": "./node_modules/.bin/webpack-dev-server --inline --config webpack.config.js --content-base examples --port 8721",
-    "demo:build": "NODE_ENV=production ./node_modules/.bin/webpack",
-    "build": "./node_modules/.bin/babel src/ --out-dir lib/ --stage 0 --loose all",
-    "lint": "./node_modules/.bin/eslint -c .eslintrc src/"
+    "test": "karma start test/karma.conf.js",
```
Comment (doug-wade) [inline package.json]: npm scripts have `node_modules/.bin` on the path by default; removing the [path separators improves portability](https://gist.github.com/domenic/2790533#being-windows-developer-friendly)

File: src/decorator.js [hunk 4]
```diff
@@ -1,22 +1,21 @@
...
-const getDisplayName = (WrappedComponent) => {
-  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
-};
+const getDisplayName = (WrappedComponent) => WrappedComponent.displayName || WrappedComponent.name || 'Component';
 
-export default (options = {}) => {
-  return function lazyload(WrappedComponent) {
-    return class LazyLoadDecorated extends Component {
-      static displayName = `LazyLoad${getDisplayName(WrappedComponent)}`;
+export default (options = {}) => function lazyload(WrappedComponent) {
```
Comment (doug-wade) [inline src/decorator.js]: This fixed a pair of eslint warnings

```
dougwade code/react-lazyload ‹master› » npm run lint

> react-lazyload@2.0.1 lint /Users/doug.wade/code/react-lazyload
> eslint -c .eslintrc src/


/Users/doug.wade/code/react-lazyload/src/decorator.js
   4:46  warning  Unexpected block statement surrounding arrow body  arrow-body-style
   8:34  warning  Unexpected block statement surrounding arrow body  arrow-body-style
  10:12  warning  Component should be written as a pure function     react/prefer-stateless-function

✖ 3 problems (0 errors, 3 warnings)
```

File: webpack.config.js [hunk 5]
```diff
@@ -27,7 +27,11 @@ module.exports = {
   module: {
     loaders: [{
       test: /\.jsx?$/,
-      loader: 'babel?stage=0&loose=all',
+      loader: 'babel',
+      query: {
```
Comment (doug-wade) [inline webpack.config.js]: I think this is readabler; its certainly writeabler

File: examples/pages/decorator.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React, { Component } from 'react';
-import { lazyload } from '../../src/';
+import lazyload from '../../src/decorator';
```
Comment (doug-wade) [inline examples/pages/decorator.js]: I may need to update the docs as well.

**Comment by doug-wade** [comment]
I had to introduce a breaking api change, changing the import for the decorator from `import {lazyloader} from 'react-lazyloader'` => `import {decorator} as lazyloader from 'react-lazyloader/decorator'`.  I should test with a reduced use case against Babel 6, check the spec, and maybe submit a pr with a broken test to babel.

**Comment by jasonslyvia** [comment]
Wow that's a lot of stuff, I need some time to dive into every piece of your comment.

File: src/index.js (hunk 1 above)
Comment (jasonslyvia) [inline src/index.js]: How about 

```javascript
// src/index.js
import decorator from './decorator';
export const lazyload = decorator;
```

API changes are always sore spot for module consumers.

File: package.json (hunk 3 above)
Comment (jasonslyvia) [inline package.json]: fair point

**Comment by jasonslyvia** [comment]
Can you squash your commit and leave out `lib/` changes for better diff?

**Comment by doug-wade** [comment]
@jasonslyvia Some great points; sorry for the long turnaround on feedback.  I think this is much more reasonable pr.  Sorry for shoving a bunch of stuff into one pr; we did a lot of iteration before we got this working internally, and I didn't want to open a PR until it was clear we'd actually be relying on it, and I could be sure it would work and be productive.

File: .babelrc [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["stage-0", "es2015"],
+  "plugins": ["transform-react-jsx", "transform-decorators-legacy"]
```
Comment (jasonslyvia) [inline .babelrc]: Why use `transform-react-jsx` plugin instead of `react` preset?

**Comment by jasonslyvia** [comment]
I think we're good to go as per this PR, is there further modification you're going to make, or should I merge it now?

**Comment by doug-wade** [comment]
I think its ready to merge!  Thanks for being so patient.

**Pull request merged**